arch/risc-v/src/bl602/bl602_systemreset.c

```diff
@@ -47,19 +53,33 @@
  *
  ****************************************************************************/
 
-static void bl602_chip_reset(uint32_t mask)
+void __attribute__((section(".tcm_code"))) bl602_chip_reset(uint32_t mask)
 {
+  /* We choose to use ROM driver */
```

Comment:
Why? The rom API does not provide proper OS locking when modifying registers which makes it less safe. This implemented the same logic.

Reply:
The ROM Driver has the same function as these commented out codes. When calling the reset code of the ROM Driver, the global interrupt needs to be turned off instead of OS-related locking. And this part of the code needs to be accessible normally when resetting, because the XIP Flash controller needs to be reset when BL602 resetting, so you can choose to put this part of the code in RAM (tcm_code section) or ROM Driver. 
We want to save RAM space, so choose to use ROM Driver and close the global interrupt to ensure that the reset process is not interrupted.